From 6c8cb966f060e7f6c813c4edc8e5db0fa6cb585d Mon Sep 17 00:00:00 2001
From: Luo Jie <quic_luoj@quicinc.com>
Date: Fri, 14 Feb 2025 17:33:36 +0800
Subject: [PATCH 31/44] arm64: dts: ipq5424: Add MDIO node

There are QCA8386 switch and two QCA8111 PHYs connected on
RDP466 devices.

Change-Id: Ia2fabc8b8f0fc8e4dcfee5d3f281a73ff3ce2977
Signed-off-by: Luo Jie <quic_luoj@quicinc.com>
---
 arch/arm64/boot/dts/qcom/ipq5424-rdp466.dts | 18 +++++++++++++
 arch/arm64/boot/dts/qcom/ipq5424.dtsi       | 28 +++++++++++++++++++++
 2 files changed, 46 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/ipq5424-rdp466.dts b/arch/arm64/boot/dts/qcom/ipq5424-rdp466.dts
index 00ea225f6..25f617277 100644
--- a/arch/arm64/boot/dts/qcom/ipq5424-rdp466.dts
+++ b/arch/arm64/boot/dts/qcom/ipq5424-rdp466.dts
@@ -195,6 +195,24 @@ &usb3 {
 	status = "okay";
 };
 
+&mdio {
+	phy0: ethernet-phy@8 {
+		compatible ="ethernet-phy-ieee802.3-c45";
+		reg = <8>;
+		reset-gpios = <&tlmm 40 GPIO_ACTIVE_LOW>;
+		reset-assert-us = <100000>;
+		reset-deassert-us = <100000>;
+	};
+
+	phy1: ethernet-phy@c {
+		compatible ="ethernet-phy-ieee802.3-c45";
+		reg = <0xc>;
+		reset-gpios = <&tlmm 28 GPIO_ACTIVE_LOW>;
+		reset-assert-us = <100000>;
+		reset-deassert-us = <100000>;
+	};
+};
+
 /*
  * The bootstrap pins for the board select the XO clock frequency that
  * supports 48 MHZ, 96 MHZ or 192 MHZ. This setting automatically
diff --git a/arch/arm64/boot/dts/qcom/ipq5424.dtsi b/arch/arm64/boot/dts/qcom/ipq5424.dtsi
index 2020b96bd..c94a5edf8 100644
--- a/arch/arm64/boot/dts/qcom/ipq5424.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq5424.dtsi
@@ -274,6 +274,18 @@ pcie3_phy: phy@fc000 {
 			status = "disabled";
 		};
 
+		mdio: mdio@90000 {
+			compatible = "qcom,ipq4019-mdio", "qcom,ipq5424-mdio";
+			reg = <0 0x00090000 0 0x64>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			pinctrl-0 = <&mdio_pins>;
+			pinctrl-names = "default";
+			clocks = <&gcc GCC_MDIO_AHB_CLK>;
+			clock-names = "gcc_mdio_ahb_clk";
+			status = "disabled";
+		};
+
 		cmn_pll: clock-controller@9b000 {
 			compatible = "qcom,ipq5424-cmn-pll";
 			reg = <0 0x0009b000 0 0x800>;
@@ -316,6 +328,22 @@ uart1_pins: uart1-state {
 				drive-strength = <8>;
 				bias-pull-up;
 			};
+
+			mdio_pins: mdio-pins {
+				mdc-state {
+					pins = "gpio22";
+					function = "mdc_mst";
+					drive-strength = <8>;
+					bias-pull-up;
+				};
+
+				mdio-state {
+					pins = "gpio23";
+					function = "mdio_mst";
+					drive-strength = <8>;
+					bias-pull-up;
+				};
+			};
 		};
 
 		gcc: clock-controller@1800000 {
-- 
2.34.1

